Author: felix
Date: 2010-05-20 11:29:39 +0200 (Thu, 20 May 2010)
New Revision: 7792

Modified:
   trunk/openvas-libraries/ChangeLog
   trunk/openvas-libraries/base/openvas_certificate_file.c
Log:
* base/openvas_certificate_file.c (openvas_certificate_file_write):
Adressed compiler warning that prevented build on some systems.
Open the file, close it directly, set contents with GLib File Utility
Functions.


Modified: trunk/openvas-libraries/ChangeLog
===================================================================
--- trunk/openvas-libraries/ChangeLog	2010-05-20 08:29:02 UTC (rev 7791)
+++ trunk/openvas-libraries/ChangeLog	2010-05-20 09:29:39 UTC (rev 7792)
@@ -1,3 +1,10 @@
+2010-05-20  Felix Wolfsteller <felix.wolfsteller at greenbone.net>
+
+	* base/openvas_certificate_file.c (openvas_certificate_file_write):
+	Adressed compiler warning that prevented build on some systems.
+	Open the file, close it directly, set contents with GLib File Utility
+	Functions.
+
 2010-05-19  Michael Wiegand <michael.wiegand at greenbone.net>
 
 	Post-release version bump.

Modified: trunk/openvas-libraries/base/openvas_certificate_file.c
===================================================================
--- trunk/openvas-libraries/base/openvas_certificate_file.c	2010-05-20 08:29:02 UTC (rev 7791)
+++ trunk/openvas-libraries/base/openvas_certificate_file.c	2010-05-20 09:29:39 UTC (rev 7792)
@@ -116,7 +116,7 @@
       g_hash_table_foreach (certs, (GHFunc) add_cert_to_file, key_file);
     }                           // (else file content is comment only)
 
-  // Write GKeyFile to filesystem.
+  // Open/Close/Create file in the desired mode.
   fd = open (filename, O_RDWR | O_CREAT | O_TRUNC, 0600);
   if (!fd)
     {
@@ -124,19 +124,29 @@
       g_key_file_free (key_file);
       return FALSE;
     }
+  if (close (fd) != 0)
+    {
+      g_key_file_free (key_file);
+      return FALSE;
+    }
 
+  // Create data to write to file.
   keyfile_data = g_key_file_to_data (key_file, &data_length, &err);
   if (err != NULL)
     {
       //show_error(_("Error exporting key file: %s"), err->message);
       g_error_free (err);
       g_key_file_free (key_file);
-      close (fd);
       return FALSE;
     }
 
-  write (fd, keyfile_data, data_length);
-  close (fd);
+  // Write data to file.
+  if (g_file_set_contents (filename, keyfile_data, data_length, &err) == FALSE)
+    {
+      g_error_free (err);
+      g_key_file_free (key_file);
+      return FALSE;
+    }
 
   g_key_file_free (key_file);

